### ISSUE TITLE
Problem with downloading water beetle data

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1033]
For the water beetle scheme Garth Foster (warehouse ID 102855) wishes to download the data for his verification role, but the download is failing:

![image](https://user-images.githubusercontent.com/16039111/112128331-3c7ffd00-8bbe-11eb-899a-d2aec8d34abf.png)

His original filter was:
https://www.brc.ac.uk/irecord/filters/admin-edit?filters_user_id=259

There have been some changes in UKSI taxonomy since that filter was created, so I have just created a new version using the scratchpad:
https://www.brc.ac.uk/irecord/filters/admin-edit?filters_user_id=10791

The new version produces considerably fewer records (c. 11,000 compared to c. 50,000 for the old one, but I think that is correct - the old one included a wider range of families). However, the new filter still fails to download.

Both filters seem to work correctly on the verification page.

The only issue I can think of is that the filters (both old and new) include a mix of family-level taxa and genus-level taxa - could that cause the download to fail?

**Comment by johnvanbreda** [comment]
@kitenetter I found a batch of 20 adjacent records where the group data in the reporting cache tables was incorrect - they had a group (activity) title but the ID was null. I've repaired these and it will take a short while before these records are updated on Elasticsearch and the download will start working again, hopefully.

I've checked the code and there is no reason why these happened. It may have been a mistake from human intervention at some point, i.e. running a back-end query which nulled the group_id in cache_occurrences_functional.
